### DISCUSSION TITLE
[API Request] Animated album art

### DISCUSSION THREAD

**Discussion by ghost** [opensubsonic/open-subsonic-api#45] [API requests]
### Type of change

API Clarification

### Proposal description

With that out of the way, many streaming services (such as TIDAL and Apple Music) now have animated cover art for [certian](https://music.apple.com/album/1440833098) [albums](https://music.apple.com/album/1373516902). These are often in the form of a GIF or mp4 and it would be nice to have some way of handling them on Subsonic. Here's an example of what my file structure looks like: 

```
Tragic Kingdom
├── 01 Spiderwebs.flac
├── 02 Excuse Me Mr_.flac
├── 03 Just a Girl.flac
├── 04 Happy Now_.flac
├── 05 Different People.flac
├── 06 Hey You.flac
├── 07 The Climb.flac
├── 08 Sixteen.flac
├── 09 Sunday Morning.flac
├── 10 Don’t Speak.flac
├── 11 You Can Do It.flac
├── 12 World Go ’Round.flac
├── 13 End It on This.flac
├── 14 Tragic Kingdom.flac
├── cover.gif
├── cover.jpg
└── cover.mp4
```

While some servers, like [gonic](https://github.com/sentriz/gonic) and [Navidrome](https://github.com/navidrome/navidrome) seem to be able to pickup GIFs it just seem to be a still frame, and (of course) no mp4 support. Animated album covers aren't necessarily need, but it would be a nice feature for those with collections that . 

### Backward compatibility

- [X] No backward compatibility impact.

### API details

A new parameter to `getCoverArt`, which could be named something such as `animated`, that controls whether or not an animated file (such as a mp4 or GIF) can be returned, this way clients can control whether or not they receive animated cover art (depending on if they support it).

### Potential issues

Clients would now have to support mp4s as a possible output to `getCoverArt` depending on the parameters that they have set.

### Alternative solutions

A new route (`getAnimatedCoverArt`) which would do the same thing as `getCoverArt` with an `animated` parameter.

**Comment by Tolriq** [comment]
You should be able to edit the title too there's an edit button at the top.

To limit impacts on current API, I would go the second route and also expose a field everywhere that match like we have `covertArt` in song we should expose an `animatedCoverArt` with a different id.

This is more efficient as it prevent making calls at client side for nothing. If we know there's an animated beforehand it's also simplify the api endpoint it should not fallback to the non animated one. It should return an error.
